bundles/com.espressif.idf.core/src/com/espressif/idf/core/build/IDFBuildConfiguration.java

```diff
@@ -451,7 +451,7 @@ private void runCmakeBuildCommand(IConsole console, IProgressMonitor monitor, IP
 
 			Instant finish = Instant.now();
 			long timeElapsed = Duration.between(start, finish).toMillis();
-			if (!monitor.isCanceled())
+			if (!monitor.isCanceled() && epm.getErrorCount() == 0)
 			{
 				project.refreshLocal(IResource.DEPTH_INFINITE, monitor);
```

Comment:
@sigmaaa is it going to be fine in terms that the indexer may not be triggered if the project is not refreshed or it will be triggered in both cases?

Reply:
Hi @alirana01,

The project will be refreshed during indexing anyway if I remember correctly.